src/generators.h

```diff
@@ -57,44 +57,21 @@ enum struct DeviceType {
 std::string to_string(DeviceType device_type);
 
 struct GeneratorParams : std::enable_shared_from_this<GeneratorParams>, LeakChecked<GeneratorParams> {
-  GeneratorParams() = default;  // This constructor is only used if doing a custom model handler vs built-in
+  GeneratorParams(const Config& config);  // This constructor is only used for internal generator benchmarks
```

Comment:
Can we remove this constructor completely? It requires us to maintain two constructors and can lead to unnecessary overhead if we update one and not the other.
Since this is only use for the benchmark code, maybe we can update the benchmark code to overwrite the config file of the model instead of exposing this constructor?

We can do this later though.

Reply:
I agree, I wanted to remove it but having all of the benchmarks require separate configs felt like a lot of work. I think after the continuous decoding change and the GeneratorParams no longer have input IDs I can simplify it further.